### PR TITLE
Continuing from #89

### PR DESCRIPTION
**Pull request by Krastanov**

### CODE REVIEW

**Comment by Krastanov** [comment]
@dmtrung14 , see the commits I have added to your work here. Could you plan to start working on the TODOs I have left in?

**Comment by dmtrung14** [comment]
> @dmtrung14 , see the commits I have added to your work here. Could you plan to start working on the TODOs I have left in?

@Krastanov I am just making sure that I get things right. There were about 5-6 TODOs that you left in the comments beside the code. Is that sufficient?

**Comment by Krastanov** [comment]
Hi, Trung! Yes, I confirm that is what I meant. Message me if any of the TODOs are troublesome.

**Comment by Krastanov** [comment]
@dmtrung14 , I rebased and squashed the commits as there was a lot of unnecessary small edits in separate commits

File: src/ecc/ECC.jl [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+function naive_syndrome_circuit(c::AbstractECC)
+    naive_sc = []
+
+    ancilla_bit = 1
+    # ancilla_qubit = code_n(c) + ancilla_bit
+    pc = parity_checks(c)
+    for check in pc
+        ancilla_qubit = code_n(c) + ancilla_bit
+        for qubit in 1: code_n(c)
+            if check[qubit] == (1,0) # TODO simplify this branch using sXCX and similar gates
```
Comment (dmtrung14) [inline src/ecc/ECC.jl:43]: is this not implemented (as in the code)?

**Pull request merged**